feat(chat): new chat tab

ice-hades opened the pull request.

ice-hades committed:
feat: new chat tab

ice-hector commented on src/navigation/Main.tsx:
```diff
@@ -291,13 +378,14 @@ const MainTabs = () => {
-          name="NewsTab"
-          component={News}
+          name="ChatTab"
+          //component={ChatTabStackNavigator}
```
Pls remove if not needed anymore

ice-hades replied:
fixed

ice-hector commented on src/navigation/components/ChatTabBar/index.tsx:
```diff
@@ -0,0 +1,79 @@
+        <View style={styles.actionsContainer}>
+          <TouchableOpacity
+            hitSlop={HIT_SLOP}
+            onPress={() => {
```
Lets create separate function/handler for this

ice-hades replied:
fixed

ice-hector commented on src/navigation/components/ChatTabBar/index.tsx:
```diff
@@ -0,0 +1,79 @@
+          </TouchableOpacity>
+          <TouchableOpacity
+            hitSlop={HIT_SLOP}
+            onPress={() => {
```
Lets create a separate function/handler for this and remove console log

ice-hades replied:
fixed

ice-hector commented on src/screens/ChatFlow/Messages/index.tsx:
```diff
@@ -0,0 +1,141 @@
+          data={displayMessages}
+          renderItem={renderItem}
+          onEndReached={() => {
+            dispatch(ChatActions.LOAD_MESSAGES_DATA.START.create());
```
That can be called multiple times and spam the redux. Theoretically, you can combine it with `onMomentumScrollBegin ` you can set some boolean to false in that method and call dispatch in `onEndReached ` when the boolean is false

ice-hades replied:
fixed

ice-hector commented on src/screens/ChatFlow/NewChatSelector/components/ChatSelectorHeader/index.tsx:
```diff
@@ -0,0 +1,60 @@
+    alignItems: 'center',
+    padding: SCREEN_SIDE_OFFSET,
+  },
+  search: {width: '100%', marginTop: rem(24)},
```
Move style to new line

ice-atlas replied:
even more, let's try to keep everything on new line whenever it is possible. Much better for git history

ice-hades replied:
fixed

ice-hector commented on src/screens/ChatFlow/NewChatSelector/index.tsx:
```diff
@@ -0,0 +1,133 @@
+        }
+        renderItem={renderItem}
+        onEndReached={() => {
+          dispatch(ChatActions.LOAD_CHAT_USERS_DATA.START.create());
```
Same here, potentially can spam the store and  be called multiple times

ice-hades replied:
fixed

ice-hades committed:
chore: fixes after review

ice-hades committed:
feat: explore top tab

ice-hades committed:
feat: channel feed

ice-atlas commented on src/navigation/components/ChatTabBarIndicator/index.tsx:
```diff
@@ -0,0 +1,90 @@
+const getTranslateX = (
+  position: Animated.AnimatedInterpolation<number>,
+  routes: Route[],
+  getTabWidth: GetTabWidth,
```
Why we can not just use plain `getTabWidth(index: number): number,` here?

ice-hades replied:
fixed

ice-atlas commented on src/navigation/components/ChatTabBarLabel/index.tsx:
```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import {StyleSheet, Text} from 'react-native';
+
+type Props = {focused: boolean; label: string};
```
let's keep it as
```
type Props = {
focused: boolean;
label: string;
};
```

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Messages/components/MessagesRow/index.tsx:
```diff
@@ -0,0 +1,169 @@
+  }
+}
+
+export function MessagesRow({messageData}: Props) {
```
I think, `Messages`, `MessagesRow` are not good name for component/screen. If it is about chats or smth like that, let's name it accordingly

ice-hades replied:
agreed to rename to ChatRow

ice-atlas commented on src/screens/ChatFlow/Messages/constants.ts:
```diff
@@ -0,0 +1,12 @@
+export const SEARCH_HIDDEN_Y = -(
+  SEARCH_INPUT_HEIGHT +
+  SEARCH_MARGIN_VERTICAL * 2 +
+  1
```
Add comment why we need this, pls

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Messages/hooks/useAnimatedSearch.ts:
```diff
@@ -0,0 +1,65 @@
+import {useDispatch, useSelector} from 'react-redux';
+
+export function useAnimatedSearch() {
+  const searchVisible = useSelector(messagesSearchVisibleSelector);
```
Do we really need this in the global state?

ice-hades replied:
decided to use router params to trigger the search instead

ice-hades replied:
use {merge: true}

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Messages/hooks/useAnimatedSearch.ts:
```diff
@@ -0,0 +1,65 @@
+import {ChatActions} from '@store/modules/Chat/actions';
+import {messagesSearchVisibleSelector} from '@store/modules/Chat/selectors';
+import * as React from 'react';
+import {useEffect} from 'react';
```
`import React, {useEffect, useCallback} from 'react';`

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Messages/hooks/useAnimatedSearch.ts:
```diff
@@ -0,0 +1,65 @@
+    },
+    (result, previous) => {
+      if (searchVisible) {
+        if (result !== previous && result > 0 && result > (previous ?? 0)) {
```
`if (searchVisible && result !== previous && result > 0 && result > (previous ?? 0))...`

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Messages/index.tsx:
```diff
@@ -0,0 +1,141 @@
+import {t} from '@translations/i18n';
+import debounce from 'lodash/debounce';
+import * as React from 'react';
+import {useEffect, useRef, useState} from 'react';
```
same about `React`. Check other places pls

ice-atlas commented on src/screens/ChatFlow/Messages/index.tsx:
```diff
@@ -0,0 +1,141 @@
+
+  const displayMessages = searchValue
+    ? messages.filter(message =>
+        message.sourceName.toLowerCase().includes(searchValue.toLowerCase()),
```
I hope, messages list gonna have pagination and we will perform search on BE via searchQuery as param for API call. This is just for UI representation for now, right?

ice-ares replied:
We’re gonna have both server&client, metadata & data search  functionality

ice-atlas commented on src/screens/ChatFlow/NewChatSelector/index.tsx:
```diff
@@ -0,0 +1,133 @@
+  const chatUsers = useSelector(chatUsersDataSelector);
+  const loading = useSelector(chatUsersLoadingSelector);
+  const [searchValue, setSearchValue] = useState('');
+  const onChangeText = debounce(setSearchValue, 600);
```
`debounce` result will be recreated on each `render`. Try to use `useMemo` with it or smth like it

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/NewChatSelector/index.tsx:
```diff
@@ -0,0 +1,133 @@
+    flex: 1,
+    overflow: 'hidden',
+    borderTopRightRadius: 30,
+    borderTopLeftRadius: 30,
```
`rem(30)`?

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/components/JoinCommunitiesBanner/index.tsx:
```diff
@@ -0,0 +1,95 @@
+  },
+  darkContainer: {
+    backgroundColor: COLORS.primaryLight,
+    borderRadius: 15,
```
`rem(15)`

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/components/JoinCommunitiesBanner/index.tsx:
```diff
@@ -0,0 +1,95 @@
+  imageContainer: {
+    width: rem(46),
+    height: rem(46),
+    borderRadius: 15,
```
rem

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/components/JoinCommunitiesBanner/index.tsx:
```diff
@@ -0,0 +1,95 @@
+    alignItems: 'center',
+    backgroundColor: COLORS.white,
+    margin: rem(14),
+    marginRight: 0,
```
add comment why we need this, pls

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/components/JoinCommunitiesBanner/index.tsx:
```diff
@@ -0,0 +1,95 @@
+  darkImageContainer: {
+    width: rem(36),
+    height: rem(36),
+    borderRadius: 12,
```
rem

ice-hades replied:
fixed

ice-atlas commented on src/screens/HomeFlow/Home/components/Header/components/NewsButton.tsx:
```diff
@@ -0,0 +1,58 @@
+
+  const dispatch = useDispatch();
+
+  const badgeCount = 1; //useSelector(NewsSelectors.getUnreadCount);
```
Why not to use real data?

ice-hades replied:
fixed

ice-atlas commented on src/store/modules/Chat/reducer/index.ts:
```diff
@@ -0,0 +1,102 @@
+
+  chatUsers: ChatUserData[];
+  hasMoreChatUsers: boolean;
+  loadingChatUsers: boolean;
```
Why do we need this in global state? `loadingMessages` & `loadingChatUsers`. Other booleans also need some sort of description.
Try to use such construction for not clear keys
```
/**
* Some description here
*/
```

ice-hades replied:
see
```
const isUserLoading = useSelector(
    isLoadingSelector.bind(null, UsersActions.GET_USER_BY_ID),
  );
  ```
  for example

ice-hades replied:
fixed

ice-atlas commented on src/store/modules/Chat/reducer/index.ts:
Also why module has name `Chat`? Is it about 1 single chat? Or `Chats`?

ice-hades replied:
fixed

ice-atlas commented on src/store/modules/Chat/sagas/index.ts:
```diff
@@ -0,0 +1,16 @@
+    takeLatest([ChatActions.LOAD_MESSAGES_DATA.START.type], loadMessagesSaga),
+    takeLatest(
+      [ChatActions.LOAD_CHAT_USERS_DATA.START.type],
+      loadChatUsersSaga,
```
`Chat users` not really nice name, IMHO. Maybe `chat members` will be better?

ice-ares replied:
Yep

ice-hades replied:
that is not some chat memebers, but a list of users you can chat with

ice-atlas commented on src/store/modules/Chat/sagas/loadChatUsersSaga.ts:
```diff
@@ -0,0 +1,57 @@
+
+function getMockedChatUser(username: string) {
+  const iconRandom = Math.floor(Math.random() * (20 - 1 + 1)) + 1;
+  const timeRandom = Math.floor(Math.random() * (12 - 1 + 1)) + 1;
```
`-1 + 1 === 0` (-_-)

ice-hades replied:
fixed

ice-atlas commented on src/store/modules/Chat/sagas/loadChatUsersSaga.ts:
```diff
@@ -0,0 +1,57 @@
+        hasMore: false,
+      }),
+    );
+    return null;
```
It's a bad idea to do like that. Check this in the `START` call place. So do not call `LOAD_CHAT_USERS_DATA.START` if requirements not met

ice-hades replied:
fixed

ice-atlas commented on src/store/modules/Chat/sagas/loadMessagesSaga.ts:
```diff
@@ -0,0 +1,123 @@
+function getMockedRandomData() {
+  const iconRandom = Math.floor(Math.random() * (20 - 1 + 1)) + 1;
+  const timeRandom = Math.floor(Math.random() * (14400 - 1 + 1)) + 1;
+  const statusRandom = Math.floor(Math.random() * (3 - 1 + 1)) + 1;
```
Same `-1+1 === 0`

ice-hades replied:
fixed

ice-atlas commented on src/store/modules/Chat/sagas/loadMessagesSaga.ts:
```diff
@@ -0,0 +1,123 @@
+        hasMore: false,
+      }),
+    );
+    return null;
```
Same here. Do not dispatch `START` then

ice-atlas commented on src/utils/string.ts:
```diff
@@ -19,3 +19,7 @@ export const stringToColor = (input: string = '') => {
 export const extractDigits = (str: string) => str.replace(/\D/g, '');
+
+export function normalizeSearchValue(searchValue: string) {
+  return searchValue.trim().toLowerCase();
```
Why we need to to do so (`toLowerCase`) on frontend side? I can understand `trim`, but all the other processing must be done on BE side (easier adjust then)

ice-atlas commented on src/utils/numbers.ts:
```diff
@@ -41,6 +42,26 @@ export function formatNumber(
+  t('filesize.kb'),
+  t('filesize.mb'),
+  t('filesize.gb'),
+  t('filesize.tb'),
```
maybe better to use it as `{{value}} kb` -->> `t('filesize.b', {value})`

ice-atlas commented on src/utils/date.ts:
```diff
@@ -19,3 +20,19 @@ export const getDurationString = (duration: Duration, numberOfUnits = 2) => {
+  return date.valueOf();
+}
+
+export function formatTimestamp({
```
why we need this function if we may just use `dayjs(timestamp).format(format);`? There no shared formatting or smth

ice-atlas commented on src/utils/date.ts:
```diff
@@ -19,3 +20,19 @@ export const getDurationString = (duration: Duration, numberOfUnits = 2) => {
+
+export function stripTimeFromTimestamp(timestamp: number): number {
+  const date = dayjs(timestamp).endOf('day');
+  return date.valueOf();
```
Why not `return dayjs(timestamp).endOf('day').valueOf()`?

ice-atlas commented on src/translations/i18n.ts:
```diff
@@ -46,11 +46,19 @@ export const tagRegex = (tag: string, isSingular = true) => {
 };
 
+export const LINK_WITH_ATTRIBUTES_REGEX = new RegExp(
+  '(\\[\\[:link.*?\\]\\][\\s\\S]+?\\[\\[\\/:link\\]\\])(?!.*\\[\\[\\[:${tag}\\])',
```
add pls comments
```
/**
* example of string "[[:link]]", etc
*/
```

ice-atlas commented on src/translations/i18n.ts:
```diff
@@ -46,11 +46,19 @@ export const tagRegex = (tag: string, isSingular = true) => {
+  '(\\[\\[:link.*?\\]\\][\\s\\S]+?\\[\\[\\/:link\\]\\])(?!.*\\[\\[\\[:${tag}\\])',
+);
+
+export const LINK_WITH_HREF_GROUP_REGEX = new RegExp(
```
and here add pls comment

ice-atlas commented on src/store/modules/Chat/types.ts:
```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: ice License 1.0
+
+export type ChatSourceType = 'channel' | 'group' | 'private';
```
`ChatSourceType` -> `ChatType`

ice-atlas commented on src/store/modules/Chat/types.ts:
```diff
@@ -0,0 +1,36 @@
+
+export type ChatSourceType = 'channel' | 'group' | 'private';
+export type ExploreDataType = 'channel' | 'group';
+export type MessageStatusType = 'sent' | 'received' | 'seen';
```
`MessageStatusType` -> `ChatMessageStatus`

ice-atlas commented on src/navigation/Main.tsx:
```diff
@@ -253,6 +286,72 @@ const TeamTabStackNavigator = () => {
+);
+
+const ExploreTabStackNavigator = () => (
+  <ExploreTabStack.Navigator screenOptions={screenOptions}>
```
Do we really need this `ExploreTabStackNavigator` && `MessagesTabStackNavigator`. I think it is enough to place there only screens under the tab

ice-atlas commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/components/PostCommentsSection/index.tsx:
```diff
@@ -0,0 +1,59 @@
+    paddingLeft: rem(6),
+    ...font(14, 17, 'medium', 'primaryLight'),
+  },
+  separator: {
```
`commonStyles.flexOne`?

ice-atlas commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/components/PostEmoji/index.tsx:
```diff
@@ -0,0 +1,34 @@
+    paddingHorizontal: CHANNEL_POST_SIDE_OFFSET,
+  },
+  text: {
+    ...font(60, 80, 'medium'),
```
Should we specify color?

ice-hades replied:
it is for an emoji

ice-atlas commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/components/PostEmojiBar/index.tsx:
```diff
@@ -0,0 +1,110 @@
+        </Pressable>
+      ))}
+      <View key={'+'} style={[styles.emojiContainer, styles.plusContainer]}>
+        <PlusIcon />
```
set size and color, it can be changed in future internally anyhow

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/components/PostFile/index.tsx:
```diff
@@ -0,0 +1,60 @@
+  iconContainer: {
+    width: rem(40),
+    height: rem(40),
+    borderRadius: 10,
```
rem

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/components/PostImagesSection/index.tsx:
```diff
@@ -0,0 +1,79 @@
+};
+
+const IMAGE_WIDTH = windowWidth - SCREEN_SIDE_OFFSET * 2;
+const IMAGE_HEIGHT = IMAGE_WIDTH / 1.673170731707317;
```
Add comment from where you get this number please or add calculations based on `(W,H)`

ice-hades replied:
fixed

ice-atlas replied:
May we use exact `W / H` or `H / W` and not approximate result?

ice-atlas commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/components/PostImagesSection/index.tsx:
```diff
@@ -0,0 +1,79 @@
+const SEPARATOR_WIDTH = rem(4);
+const SMALL_IMAGE_WIDTH =
+  (windowWidth - SCREEN_SIDE_OFFSET * 2 - SEPARATOR_WIDTH) / 2;
+const SMALL_IMAGE_HEIGHT = SMALL_IMAGE_WIDTH / 1.1643835616438356;
```
and here

ice-hades replied:
fixed

ice-atlas replied:
same here

ice-atlas commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/components/PostReplySection/components/ReplyImagePreview/index.tsx:
```diff
@@ -0,0 +1,57 @@
+
+const styles = StyleSheet.create({
+  container: {
+    borderRadius: 6,
```
rem

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/components/PostReplySection/components/ReplyTextPreview/index.tsx:
```diff
@@ -0,0 +1,59 @@
+}
+
+const styles = StyleSheet.create({
+  textContainer: {
```
`commonStyles.flexOne`

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/components/PostReplySection/index.tsx:
```diff
@@ -0,0 +1,74 @@
+        <View
+          style={[
+            styles.replyIconContainer,
+            {top: containerHeight / 2 - REPLY_ICON_SIZE / 2},
```
```
{
  top: (containerHeight - REPLY_ICON_SIZE) / 2,
},
```

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/components/PostReplySection/index.tsx:
```diff
@@ -0,0 +1,74 @@
+  },
+  container: {
+    backgroundColor: COLORS.feedBackground,
+    borderRadius: 10,
```
rem

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/components/PostReplySection/index.tsx:
```diff
@@ -0,0 +1,74 @@
+  },
+  replyIconContainer: {
+    position: 'absolute',
+    top: 0,
```
Why do we need this? (add comment pls)

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/index.tsx:
```diff
@@ -0,0 +1,47 @@
+  getPostData: (postId: number) => ChannelPostData | null;
+};
+
+export function ChannelPost({postData, getPostData}: Props) {
```
Can we use `postId` or smth like this? And get all the data from store then. Easier to react on changes and do not need to pass whole object down the hierarchy

ice-hades replied:
let's discuss

ice-atlas commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/index.tsx:
```diff
@@ -0,0 +1,47 @@
+
+const styles = StyleSheet.create({
+  container: {
+    borderRadius: 16,
```
rem

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelSectionHeader/index.tsx:
```diff
@@ -0,0 +1,45 @@
+    flexDirection: 'row',
+  },
+  container: {
+    borderRadius: 20,
```
rem

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Channel/hooks/useLoadChannelData.ts:
```diff
@@ -0,0 +1,94 @@
+      const newPosts = await fetchChannelPosts({
+        channelId,
+        offset: offsetRef.current,
+        fetchSize: FETCH_SIZE,
```
Is it `limit`?

ice-hades replied:
yes

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Channel/index.tsx:
```diff
@@ -0,0 +1,47 @@
+  headerContainer: {
+    backgroundColor: COLORS.white,
+    width: '100%',
+    top: 0,
```
Why do we need this? Add comment pls

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Explore/components/ExploreFilterTile/index.tsx:
```diff
@@ -0,0 +1,112 @@
+function getFilterTypeIcon(filterType: ExploreFilterType) {
+  switch (filterType) {
+    case 'all':
+      return <InfiniteIcon color={COLORS.white} />;
```
specify size

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Explore/components/ExploreFilterTile/index.tsx:
```diff
@@ -0,0 +1,112 @@
+    case 'groups':
+      return styles.groupsBackground;
+    default:
+      return {};
```
why not `null`?

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Explore/components/ExploreFilterTile/index.tsx:
```diff
@@ -0,0 +1,112 @@
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: COLORS.secondaryFaint05opacity,
+    borderRadius: 16,
```
rem

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Messages/hooks/useAnimatedSearch.ts:
```diff
@@ -0,0 +1,67 @@
+import {getSearchVisibleSelector} from '@store/modules/Chat/selectors';
+import {ChatDataType} from '@store/modules/Chat/types';
+import * as React from 'react';
+import {useEffect} from 'react';
```
`import React, {useEffect, useCallback} from 'react';`

ice-hades replied:
fixed

ice-atlas commented on src/store/modules/Chat/sagas/loadChatDataSaga.ts:
```diff
@@ -0,0 +1,24 @@
+  const {dataType} = action.payload;
+  switch (dataType) {
+    case 'chats':
+      yield call(loadMessagesSaga, action);
```
Let's keep things simple. Implement separate action for every saga

ice-hades replied:
let's discuss

ice-hades replied:
@ice-orion did you think about it? what's your opinion?

ice-atlas commented on src/store/modules/Chat/sagas/loadExploreDataSaga.ts:
```diff
@@ -0,0 +1,142 @@
+type Actions = ReturnType<typeof ChatActions.LOAD_CHAT_DATA.START.create>;
+
+function getMockedRandomData() {
+  const iconRandom = Math.floor(Math.random() * (20 - 1 + 1)) + 1;
```
`- 1 + 1`

ice-hades replied:
fixed

ice-hades committed:
feat: channel post highlight

ice-atlas commented on src/constants/animations.ts:
```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: ice License 1.0
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+// @ts-ignore
```
Why? Add comment here

ice-hades replied:
fixed

ice-hades committed:
feat: added context emoji bar

ice-hades committed:
chore: fixed review comments

ice-orion commented on src/navigation/Main.tsx:
```diff
@@ -374,11 +488,27 @@ export function MainNavigator() {
+      <MainStack.Screen
+        name="EmojiSelector"
+        component={EmojiSelector}
+        options={{...modalOptions, animation: 'none'}}
```
Let's place the screen component from `src/screens/EmojiSelector` to `src/screens/Modals/EmojiSelector` since it's a modal

ice-hades replied:
fixed

ice-orion commented on src/App.tsx:
```diff
@@ -10,6 +11,8 @@ import {SafeAreaProvider} from 'react-native-safe-area-context';
 import {Provider} from 'react-redux';
 import {PersistGate} from 'redux-persist/integration/react';
 
+initEmojiData();
```
This function takes a raw data from `data.ts`, splits it to chunks and generates `sections` object, right?
Can we write down the generated sections to that `data.ts` instead of the raw data once we have it, so we don't need to run this function every time we launch the app?
Currently instead of slowing down the first emoji modal render we slow down every app launch that is even worse.

ice-hades replied:
fixed

ice-orion commented on src/screens/EmojiSelector/utils.ts:
```diff
@@ -0,0 +1,111 @@
+} from '@screens/EmojiSelector/type';
+import {normalizeSearchValue} from '@utils/string';
+
+export function splitArrayIntoChunks(array: EmojiData[]): EmojiData[][] {
```
Can't you use `flexWrap` - https://yogalayout.com/docs/flex-wrap/ to render all these emojis in multiple rows instead of splitting them manually?

ice-hades replied:
resolved

ice-orion commented on src/navigation/Main.tsx:
```diff
@@ -253,6 +300,72 @@ const TeamTabStackNavigator = () => {
+          },
+        },
+      }}
+      tabBar={ChatTabBarComponent}>
```
Do we need to use a wrapper here? Can't we pass `ChatTabBar` as is?
```
tabBar={ChatTabBar}>
```

ice-hades replied:
fixed

ice-orion commented on src/navigation/components/ChatTabBar/index.tsx:
```diff
@@ -0,0 +1,93 @@
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: COLORS.white,
+    overflow: 'visible',
```
isn't it a default value?

ice-hades replied:
fixed

ice-orion commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/components/PostEmojiBar/index.tsx:
```diff
@@ -0,0 +1,102 @@
+          ) : null}
+        </Pressable>
+      ))}
+      <Pressable
```
Any reason why not our custom `Touchable`? Here in all other places

ice-hades replied:
fixed

ice-orion commented on src/screens/ChatFlow/Channel/components/ChannelFeed/index.tsx:
```diff
@@ -0,0 +1,125 @@
+    );
+  };
+
+  const renderSectionHeader = ({
```
SectionList is a pure component, so it's better keep all the passed props shallow equal -> wrap `renderItem` and `renderSectionHeader` in `useCallback`

ice-hades replied:
fixed

ice-orion commented on src/screens/ChatFlow/Channel/hooks/useLoadChannelData.ts:
```diff
@@ -0,0 +1,95 @@
+        offsetRef.current = 0;
+      }
+      setLoading(true);
+      const newPosts = await fetchChannelPosts({
```
It is going to be in a module eventually, isn't it?

ice-hades replied:
discussed

ice-hades replied:
will do it in a next pr

ice-hades committed:
chore: changes after review

ice-hector commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/components/PostCommentsSection/components/CommentatorsPreview/index.tsx:
```diff
@@ -0,0 +1,66 @@
+      {remainingCommentators ? (
+        <View style={styles.remainingContainer}>
+          <Text style={styles.text}>
+            +{formatNumber(remainingCommentators, {notation: 'compact'})}
```
What about RTL support? Or will be good?

ice-hades replied:
yes, should be fine. Text inverts the whole content on RTL

ice-hector commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/components/PostEmojiBar/hooks/useEmojiBarHandlers.ts:
```diff
@@ -0,0 +1,63 @@
+    ...emojis,
+    [emojiToSwitch]: {
+      counter: isLiked ? emojiData.counter - 1 : emojiData.counter + 1,
+      liked: !isLiked,
```
`liked: !isLiked` ? its something wrong with naming here

ice-hades replied:
it is a reaction to a press on an emoji. so we invert the liked state

ice-orion committed:
chore: improve ts, all url polyfill (#62)

* Some `ts-ignores` are fixed, some just changed to something less
dangerous:
E.g. unlike @ts-ignore, @ts-expect-error throws an error on ts-lint
check if error is not there anymore. So at some point the types for
`Animated.FlatList` will be fixes, we'll know about it and remove the
comment completely

* Use `react-native-url-polyfill` instead of custom `url-parse` lib
since polyfill uses build in react-native modules to deal with urls
instead of parsing using regexps in `url-parse`. And it is easier to
switch from a polyfill if at some point RN adds the support out of the
box

* Fix the issue when on other user's profile the username of the main
user is displayed

ice-hector commented on src/screens/ChatFlow/Channel/components/ChannelFeed/ChannelPost/index.tsx:
```diff
@@ -0,0 +1,80 @@
+  darkMode?: boolean;
+};
+
+export function ChannelPost({
```
Can we pass just the postData id? And use this id and selectors inside other components to where you pass it (PostImagesSection, PostInfoSection, PostEmoji, etc). 
Also as for `getPostData, updatePostData, deletePostData` we shouldn't pass them deeply to components or to navigation params. Just create a hook that will have the implementation of those methods, and init it with the postData.id (and we are returning that you need only postData id to pass here) and use that hook method directly where you need them inside the components

ice-hades replied:
yes, we decided I do it on the next pr to this FB

ice-hector commented on src/screens/ChatFlow/Channel/components/ChannelPostHighlight/components/ContextMenu/index.tsx:
```diff
@@ -0,0 +1,130 @@
+              color={COLORS.primaryDark}
+            />
+          }
+          action={() => {}}
```
Please handle action

ice-hades replied:
it is work in progress. this action is stabbed for now

ice-hector commented on src/screens/ChatFlow/Channel/components/ChannelPostHighlight/components/ContextMenu/index.tsx:
```diff
@@ -0,0 +1,130 @@
+              color={COLORS.primaryDark}
+            />
+          }
+          action={() => {
```
Lets move to separate function

ice-hades replied:
fixed

ice-hector commented on src/screens/ChatFlow/Channel/components/ChannelPostHighlight/components/ContextMenu/index.tsx:
```diff
@@ -0,0 +1,130 @@
+        <ContextMenuAction
+          title={'Copy link'}
+          icon={<CopyLinkIcon color={COLORS.primaryDark} />}
+          action={() => {
```
Same here

ice-hades replied:
fixed

ice-hector commented on src/screens/ChatFlow/Channel/components/ChannelPostHighlight/components/ContextMenu/index.tsx:
```diff
@@ -0,0 +1,130 @@
+        <ContextMenuAction
+          title={'Report'}
+          icon={<ReportIcon color={COLORS.primaryDark} />}
+          action={() => {}}
```
Will these and below actions be handled in scope of this PR or other one?

ice-hades replied:
yes. this PR is in the FB not to the main

ice-hector commented on src/screens/ChatFlow/Messages/components/ChatRow/index.tsx:
```diff
@@ -0,0 +1,93 @@
+          </View>
+          {messageData.unreadMessages > 1 ? (
+            <View style={styles.messagesCounterContainer}>
+              <Text style={styles.messagesCounterText}>{3}</Text>
```
Is it hardcoded for now or is it the requirement to put 3 here if unread messages are greater than 1?

ice-hades replied:
yes, stabbed data here

ice-atlas commented on src/store/modules/EmojiHistory/actions/index.ts:
```diff
@@ -0,0 +1,11 @@
+import {createAction} from '@store/utils/actions/createAction';
+
+const ADD_EMOJI_TO_HISTORY = createAction('ADD_EMOJI_TO_HISTORY', {
+  STATE: (emoji: string) => ({emoji}),
```
I may suggest to use `ACTION_NAME: (payload: {emoji: string}) => payload` whenever it is possible. Because then easier to read code and add new params (just 1 place [type] instead of 2 [type, result object])

ice-hades replied:
fixed

ice-atlas commented on src/store/modules/EmojiHistory/selectors/index.ts:
```diff
@@ -0,0 +1,6 @@
+import {RootState} from '@store/rootReducer';
+
+export const emojiHistorySelector = (state: RootState) =>
+  state.emojiHistory.emojiHistory;
```
Let's improve this somehow. `state.NAME.NAME` is not really nice naming :( 
Maybe `state.emojiHistory.latestEmojis` or `state.emojiHistory.items`

ice-hades replied:
state.emojiHistory.history is fine?

ice-atlas replied:
I do not really like this variant, because it is hard to understand what's inside of it and purpose of it. And same duplication of `history`

ice-hades replied:
fixed

ice-atlas commented on src/screens/HomeFlow/Home/components/Header/components/NewsButton.tsx:
```diff
@@ -0,0 +1,59 @@
+  badge: {
+    position: 'absolute',
+    top: -6,
+    right: -5,
```
rem

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/components/ChatTabRow/index.tsx:
```diff
@@ -0,0 +1,137 @@
+          <Avatar
+            uri={icon}
+            size={rem(48)}
+            borderRadius={15}
```
rem

ice-hades replied:
fixed

ice-hades committed:
chore: ui fixes

ice-hades committed:
chore: fixed review comments

ice-atlas commented on src/screens/ChatFlow/Messages/hooks/useAnimatedSearch.ts:
```diff
@@ -0,0 +1,63 @@
+      ) {
+        translateY.value = withTiming(SEARCH_HIDDEN_Y);
+        searchVisibleSharedValue.value = 0;
+        // runOnJS(setSearchBarHidden)();
```
If do not need this anymore just remove or add comment

ice-hades replied:
fixed

ice-atlas commented on src/screens/ChatFlow/Messages/index.tsx:
`Messages` is not right name for this screen. We present a list of `chats` here, so maybe make sense to rename folder and screen

ice-hades replied:
disagree. it is named after the tab name on the UI. we can discuss with others if you insist

ice-hades replied:
fixed

ice-atlas commented on src/store/rootReducer.ts:
```diff
@@ -43,6 +46,7 @@ export const rootReducer = combineReducers({
   notifications: notificationsReducer,
   users: usersReducer,
   statusNotice,
+  chat: chatReducer,
```
`chats: chatsReducer`

ice-hades replied:
fixed

ice-atlas commented on src/screens/Modals/EmojiSelector/type.ts:
```diff
@@ -0,0 +1,30 @@
+  | 'search_results';
+
+export type EmojiData = {
+  keywords: string[];
```
How will we manage this for different locales/languages?

ice-hades replied:
we will need to add localised keywords for every supported language

ice-hades committed:
chore: changes after review
